lua/telescope/pickers.lua

```diff
@@ -1054,6 +1130,7 @@ function Picker:get_result_processor(find_id, prompt, status_updater)
   end
 end
 
+-- TODO: document what the purpose of this is
```

Comment:
Runs after all elements are scored/processed. Handles things like, what should happen with selection index:
- `reset` will set the selection to row 1 (or `opts.default_selection_index`) back
- `closest` sets it to index 1 if prompt != "" else `opts.default_selection_index` ( `builtin.buffers` make use of `default_selection_index` with its `opts.sort_last_used` option)
- row tries to keep the row (might be broken, dont wanna blame tj but i think he broke it some time ago and i was too lazy to fix it) i reported it here: https://github.com/nvim-telescope/telescope.nvim/pull/293#issuecomment-757136468 he ignored me (as always, he is just above my pay grade :rofl:, if it isn't obvious i am memeing right now)
- follow tries to follow the currently selected entry

other things that happen in the returned method is:
- status_updater, update the n/m virtual text thingy
- set `current_line` so some actions can make use of it
- clear extra rows
- do highlighting
- call finish on sorter